js/ui/payg.js

```diff
@@ -740,7 +744,10 @@ class PaygNotifier extends GObject.Object {
         Main.messageTray.add(source);
 
         // by default, this notification is for early entry of an unlock keycode
-        let messageText = NOTIFICATION_EARLY_CODE_ENTRY_TEXT;
+        let codeLength = Main.paygManager.codeLength
+        let messageText = Gettext.ngettext(NOTIFICATION_EARLY_CODE_ENTRY_TEXT_1,
+                                           NOTIFICATION_EARLY_CODE_ENTRY_TEXT_N,
+                                           codeLength).format(codeLength)
```

Comment:
Looks like semicolons might be missing at the end of the lines here.

Also, should the string format placeholder be `%s` or should it be `%u`? I’m not sure how `str.format()` works in JavaScript.

Reply:
> Looks like semicolons might be missing at the end of the lines here.
> 

Good catch! It seems they are not required by the language and perhaps a coding-style preference, as the code worked without them. In any case, we should add them here.

> Also, should the string format placeholder be `%s` or should it be `%u`? I’m not sure how `str.format()` works in JavaScript.

I thought the same when I first wrote this, but `%u` gave me a runtime error. Then looking around the file I found other places using `%s` for integers, ex., `Gettext.ngettext('%s second', '%s seconds', seconds).format(Math.floor(seconds));`, so I followed that.

Reply:
If you are morbidly curious about JavaScript's relationship to the noble semicolon: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#automatic_semicolon_insertion

Reply:
:exploding_head: Not sure I can handle living in the future.